csrc/scheduler/matmul.cpp

```diff
@@ -995,9 +1021,24 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   acr->axis(-1)->parallelize(ParallelType::Vectorize);
   bcr->axis(-1)->parallelize(ParallelType::Vectorize);
 
-  //  0  1  2  3  4   5   6   7  8  9  10 11
-  // [B Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
-  if (num_batch_dims != 0) {
+  // Parallelization strategy:
+  //  with splitk:
+  // nbatch +   1   2   3   4    5    6    7    8     9    10    11   12
+  //      -13 -12 -11 -10  -9   -8   -7   -6   -5    -4    -3    -2   -1
+  // [..., Mo, No, Kf, Ko, Kw, Mwo, Nwo, Mwi, Nwi, MNi1, MNi2, MNi3,  Ki]
+  //  (iS) iBx iBy rBz rS  rS  iTz  iTy   iS   iS  iMMA   iTx  iMMA rMMA
+  //
+  //  without splitk:
+  // nbatch +   1       2   3    4    5    6    7     8     9    10   11
+  //      -12 -11     -10  -9   -8   -7   -6   -5    -4    -3    -2   -1
+  // [..., Mo, No,     Ko, Kw, Mwo, Nwo, Mwi, Nwi, MNi1, MNi2, MNi3,  Ki]
+  // (iBz) iBx iBy     rS  rS  iTz  iTy   iS   iS  iMMA   iTx  iMMA rMMA
+
+  // When we have both batch dims and splitk, parallelize splitk only.
```

Comment:
Looks like we are now supporting split-k of batched matmul? Could you add a test for that? It might not be performant, but functionally should work.

Reply:
Thanks for the suggestion. I added this test and discovered a bug in how I was transposing for BMMs. It's fixed now with more comments in `test/utils.cpp`.